yang/frr-ospfd.yang

```diff
@@ -0,0 +1,957 @@
+          type uint16 {
+            range "3..65535";
+          }
+          units "seconds";
+          description
+            "Time between retransmitting lost link state advertisements.";
+        }
+
+        leaf transmit-delay {
+          type uint16 {
+            range "1..65535";
+          }
+          units "seconds";
+          description
+            "Link state transmit delay.";
+        }
+      }
+    }
+  }
+
+  grouping area-groupings {
+    container areas {
+      list area {
+        key "area-id";
+        description
+          "A list of area objects";
+        leaf area-id {
+          type ospf-area-id;
+        }
+
+        container authentication {
+          presence "Enable authentication";
+          leaf message-digest {
+            type boolean;
+            description
+              "Use message-digest authentication";
+          }
+        }
+
+        leaf default-cost {
+          type uint32 {
+            range "0..16777215";
+          }
+          description
+            "Advertised default summary cost";
+        }
+
+        leaf export-list {
+          type access-list-ref;
+          description
+            "Filter for networks announced to other areas.";
+        }
+
+        leaf import-list {
+          type access-list-ref;
+          description
+            "Filter for networks from other areas announced to the specified one.";
+        }
+
+        container filter-list {
+          leaf prefix {
+            type plist-ref;
+            description
+              "Filter networks between OSPF areas.";
+          }
+
+          leaf in {
+            type boolean;
+          }
+
+          leaf out {
+            type boolean;
+          }
+        }
+
+        uses nssa-grouping;
+
+        uses range-grouping;
+
+        uses stub-grouping;
+
+        uses shortcut-grouping;
+
+        uses virtual-link-group;
+      }
+    }
+  }
+
+  /* router ospf attributes */
+  grouping route-ospf-leaf-attributes {
+    leaf auto-cost-reference-bandwidth {
+      type uint32 {
+        range "1..4294967";
+      }
+      units "Mbits";
+      description
+        "The reference bandwidth in terms of Mbits per second.";
+    }
+
+    leaf capability-opaque {
+      type boolean;
+      description
+        "Opaque LSA.";
+    }
+
+    leaf compatible-rfc1583 {
+      type boolean;
+      description
+        "Compatible with RFC 1583.";
+    }
+
+    leaf default-metric {
+      type uint32 {
+        range "0..16777214";
+      }
+      description
+        "Metric of redistributed routes.";
+    }
+
+    leaf write-multiplier {
+      type uint8 {
+        range "1..100";
+      }
+      description
+        "Maximum number of interfaces serviced per write.";
+    }
+
+    container router-info {
+      leaf as {
+        type boolean;
+        description
+          "Enable the Router Information functionality with AS flooding scope.";
+      }
+
+      leaf area {
```

Comment:
So this container is modelling the "router-info <as|area [A.B.C.D]>" command. Since this command accepts either "as" or "area [A.B.C.D]" (and not both at the same time), it'd be better to use a "choice" statement to prevent both leafs from being present at the same time (either that or add some validation checks in the NB callbacks).